src/Catalog/Dnx/DnxMaker.cs

```diff
@@ -0,0 +1,132 @@
+{
+    public class DnxMaker
+    {
+        private readonly StorageFactory _storageFactory;
+
+        public DnxMaker(StorageFactory storageFactory)
+        {
+            if (storageFactory == null)
+            {
+                throw new ArgumentNullException(nameof(storageFactory));
+            }
+
+            _storageFactory = storageFactory;
+        }
+
+        public async Task AddPackage(Stream nupkgStream, string nuspec, string id, string version, CancellationToken cancellationToken)
+        {
+            Storage storage = _storageFactory.Create(id);
+
+            await SaveNuspec(storage, id, version, nuspec, cancellationToken);
+            await SaveNupkg(nupkgStream, storage, id, version, cancellationToken);
+            await UpdateMetadata(storage, versions => versions.Add(NuGetVersion.Parse(version)), cancellationToken);
+        }
+
+        public async Task DeletePackage(string id, string version, CancellationToken cancellationToken)
+        {
+            Storage storage = _storageFactory.Create(id);
+
+            await UpdateMetadata(storage, versions => versions.Remove(NuGetVersion.Parse(version)), cancellationToken);
+            await DeleteNuspec(storage, id, version, cancellationToken);
+            await DeleteNupkg(storage, id, version, cancellationToken);
+        }
+
+        private async Task SaveNuspec(Storage storage, string id, string version, string nuspec, CancellationToken cancellationToken)
+        {
+            string relativeAddress = string.Format("{1}/{0}.nuspec", id, version);
+            Uri nuspecUri = new Uri(storage.BaseAddress, relativeAddress);
+            await storage.Save(nuspecUri, new StringStorageContent(nuspec, "text/xml", "max-age=120"), cancellationToken);
+        }
+
+        async Task UpdateMetadata(Storage storage, Action<HashSet<NuGetVersion>> updateAction, CancellationToken cancellationToken)
+        {
+            string relativeAddress = "index.json";
+
+            Uri resourceUri = new Uri(storage.BaseAddress, relativeAddress);
+            HashSet<NuGetVersion> versions = GetVersions(await storage.LoadString(resourceUri, cancellationToken));
+            updateAction(versions);
+            List<NuGetVersion> result = new List<NuGetVersion>(versions);
+
+            if (result.Any())
+            {
+                // Store versions (sorted)
+                result.Sort();
+                await storage.Save(resourceUri, CreateContent(result.Select((v) => v.ToString())), cancellationToken);
+            }
+            else
+            {
+                // Remove versions file if no versions are present
+                if (storage.Exists(relativeAddress))
+                {
+                    await storage.Delete(resourceUri, cancellationToken);
+                }
+            }
+        }
+
+        private static HashSet<NuGetVersion> GetVersions(string json)
+        {
+            HashSet<NuGetVersion> result = new HashSet<NuGetVersion>();
```

Comment:
its OK to use var in a narrow scope like this with an explicit assignment - there is no loss of readability